### PR TITLE
esa_s2 bands and png new name format

### PR DESCRIPTION
**Pull request by toshaklg**
How it looks like:
image

### CODE REVIEW

File: vsm/lib/raster/esa_s2.cpp [hunk 1]
```diff
@@ -378,9 +378,9 @@ bool ESA_S2_Image::splitJP2(const std::filesystem::path &path_in, const std::fil
 
 			std::filesystem::create_directories(ss_path_out.str());
 
-			ss_path_out_nc << ss_path_out.str() << "bands.nc";
-			ss_path_out_png << ss_path_out.str() << path_in.stem().string() << "_" << tx0 << "_" << ty0 << ".png";
-
+			ss_path_out_nc << ss_path_out.str() << path_in.stem().string().substr(0, path_in.stem().string().find_last_of("_")) << "_tile_" << xi << "_" << yi << ".nc";
```
Comment (indrek-sunter) [inline vsm/lib/raster/esa_s2.cpp]: Perhaps we could have a function which performs
`path_in.stem().string().substr(0, path_in.stem().string().find_last_of("_"))`

File: vsm/lib/raster/esa_s2.cpp [hunk 2]
```diff
@@ -378,9 +378,9 @@ bool ESA_S2_Image::splitJP2(const std::filesystem::path &path_in, const std::fil
...
+			ss_path_out_png << ss_path_out.str() << path_in.stem().string() << "_"<< "tile_" << xi << "_" << yi << ".png";
```
Comment (indrek-sunter) [inline vsm/lib/raster/esa_s2.cpp]: I would prefer to have spaces around << operators.

File: vsm/lib/raster/esa_s2.cpp [hunk 3]
```diff
@@ -493,8 +493,8 @@ bool ESA_S2_Image::splitTIF(const std::filesystem::path &path_in, const std::fil
 
 			std::filesystem::create_directories(ss_path_out.str());
 
-			ss_path_out_nc << ss_path_out.str() << "bands.nc";
-			ss_path_out_png << ss_path_out.str() << path_in.stem().string() << "_" << tx0 << "_" << ty0 << ".png";
+			ss_path_out_nc << ss_path_out.str() << path_in.stem().string().substr(0, path_in.stem().string().find_last_of("_")) << "_tile_" << xi << "_" << yi << ".nc";
+			ss_path_out_png << ss_path_out.str() << path_in.stem().string() << "_"<< "tile_" << xi << "_" << yi << ".png";
```
Comment (indrek-sunter) [inline vsm/lib/raster/esa_s2.cpp]: Same here, would prefer spaces around <<

File: vsm/lib/raster/esa_s2.cpp [hunk 4]
```diff
@@ -608,8 +608,8 @@ bool ESA_S2_Image::splitPNG(const std::filesystem::path &path_in, const std::fil
...
```
Comment (indrek-sunter) [inline vsm/lib/raster/esa_s2.cpp]: And here, would prefer spaces around <<

**Review by indrek-sunter — CHANGES_REQUESTED** [review]
Perhaps a few minor changes

File: vsm/include/raster/esa_s2.hpp [hunk 5]
```diff
@@ -107,6 +107,11 @@ class ESA_S2_Image {
 		 */
 		void set_resampling_method(const std::string &m);
 
+		/**
+		 * Gets index, first date and bands for files' names.
+		 */
+		std::string index_date_band_extractor(const std::filesystem::path &path);
```
Comment (indrek-sunter) [inline vsm/include/raster/esa_s2.hpp]: I think `extract_index_date_band` would be better.

**Review by indrek-sunter — CHANGES_REQUESTED** [review]
Also, perhaps we could also make the necessary changes in the rasterization and other converters?

File: vsm/lib/raster/esa_s2.cpp [hunk 6]
```diff
@@ -378,9 +378,9 @@ bool ESA_S2_Image::splitJP2(const std::filesystem::path &path_in, const std::fil
 
 			std::filesystem::create_directories(ss_path_out.str());
 
-			ss_path_out_nc << ss_path_out.str() << "bands.nc";
-			ss_path_out_png << ss_path_out.str() << path_in.stem().string() << "_" << tx0 << "_" << ty0 << ".png";
-
+			ss_path_out_nc << ss_path_out.str() << extract_index_date_band(path_in) << "_tile_" << xi << "_" << yi << ".nc";
+			ss_path_out_png << ss_path_out.str() << path_in.stem().string() << "_" << "tile_" << xi << "_" << yi << ".png";
+
```
Comment (indrek-sunter) [inline vsm/lib/raster/esa_s2.cpp]: Minor: Empty line :)

**Comment by toshaklg** [comment]
The folder with segments.ai now looks like:
segmentsai

**Review by indrek-sunter — APPROVED** [review]

**Pull request merged**